resources/resources.go

```diff
@@ -51,6 +83,20 @@ func MakeResource(source unstructured.Unstructured) Resource {
 	return res
 }
 
+func MakeQuery(typeSpecimen client.Object, scheme runtime.Scheme, namespaces []string, uid types.UID) (ResourceCounterQuery, error) {
+	gvk, _, err := scheme.ObjectKinds(typeSpecimen)
+	if err != nil {
+		return ResourceCounterQuery{}, err
+	}
+	return ResourceCounterQuery{
+		Namespaces: namespaces,
+		//This creeps me out and I do not like it
+		//Assuming the first entry is right feels very fly by night
+		GVK:       gvk[0],
```

Comment:
We also need to protect against the fact that we could have a nil length without an err. This would blow up here without it.

Reply:
There is the possibility that the specimen is provided but isn't present in the schema, I think _that_ would cause the nil length. - Thoughts?

Reply:
I'm not sure - I assumed passing something that isn't in the schema to scheme.ObjectKinds would result in err not nil. I'll do some testing and see what I end up with.

Reply:
Read the implementation https://github.com/kubernetes/apimachinery/blob/master/pkg/runtime/scheme.go#L243 and it looks like we can't ever get nil there. But I decided to test a bit for sure. If you pass in a type that isn't registered to the scheme you get a not registered error, so I think we're good here. I put my findings into two new unit tests: one for MakeQuery with a registered type and one without. Thanks!

Reply:
OK no worries